### PR TITLE
Re 74 default date value

### PR DESCRIPTION
**Pull request by nickkhall**
Added default date functionality in React-Formio
-- Uses Moment.js functions

### CODE REVIEW

File: src/Partials/Components/datetime.jsx [hunk 1]
```diff
@@ -29,9 +55,5 @@ module.exports = React.createClass({
         onChange={this.onChangeDatetime.bind(null, index)}
         />
     );
-  },
```
Comment (randallknutson) [inline src/Partials/Components/datetime.jsx]: Is there a reason you removed this?

File: src/Partials/Components/datetime.jsx [hunk 2]
```diff
@@ -10,10 +10,36 @@ module.exports = React.createClass({
...
+    if (isNaN(dateVal)) {
+      dateVal = '';
+    }
+
+    return dateVal;
   },
   onChangeDatetime: function(index, value, str) {
-    this.setValue(value, index);
+    this.setValue(
+      this.parseMomentValue(value),
```
Comment (randallknutson) [inline src/Partials/Components/datetime.jsx]: Where is this.parseMomentValue defined? I'd like to do this the same way we did in angular where the eval statement in line 25 has moment and Date available to be evaled.

File: src/Partials/Components/datetime.jsx [hunk 3]
```diff
@@ -10,10 +10,36 @@ module.exports = React.createClass({
...
+      this.props.component.defaultDate.length === 0
+    ) {
+      return '';
+    }
+    const { defaultDate } = this.props.component;
+
+    var dateVal = new Date(defaultDate);
+    if (isNaN(dateVal.getDate())) {
+      try {
+        dateVal = new Date(eval(defaultDate));
```
Comment (randallknutson) [inline src/Partials/Components/datetime.jsx]: Shouldn't we make moment available to the eval statement?

File: src/Partials/Components/datetime.jsx (hunk 1 above)
Comment (nickkhall) [inline src/Partials/Components/datetime.jsx]: Hmm, not sure why that was removed, it's in my code, just with no comma.

File: src/Partials/Components/datetime.jsx (hunk 3 above)
Comment (nickkhall) [inline src/Partials/Components/datetime.jsx]: Well, I'm pretty sure it is available to the eval function, it's being imported at the top of the file.

It's also like that in the Angular version.
https://github.com/formio/ngFormio/blob/master/src/components/datetime.js#L29

File: src/Partials/Components/datetime.jsx (hunk 2 above)
Comment (nickkhall) [inline src/Partials/Components/datetime.jsx]: I think I accidentally left that in there.

File: src/Partials/Components/datetime.jsx (hunk 1 above)
Comment (randallknutson) [inline src/Partials/Components/datetime.jsx]: I was actually referring to the following several lines. You took out the getValueDisplay function for some reason.

**Pull request closed without merging**